addons/portal/models/portal_mixin.py

```diff
@@ -71,18 +71,18 @@ def _notify_get_groups(self, msg_vals=None):
             local_msg_vals.update(customer.signup_get_auth_param()[customer.id])
             access_link = self._notify_get_action_link('view', **local_msg_vals)
 
-            new_group = [
-                ('portal_customer', lambda pdata: pdata['id'] == customer.id, {
+            new_group = (
+                'portal_customer', lambda pdata: pdata['id'] == customer.id, {
                     'has_button_access': False,
                     'button_access': {
                         'url': access_link,
                     },
                     'notification_is_customer': True,
-                })
-            ]
-        else:
-            new_group = []
-        return new_group + groups
+                }
+            )
+            # Insert the new group between user and portal
+            groups.insert(1, new_group)
```

Comment:
Don't change that in stable. Moreover it does not seem more correct, being the customer means seeing the frontend version to accept and pay for example. So I would keep this code like it is.